weave/util.py

```diff
@@ -21,25 +37,36 @@ def init_sentry():
     except ImportError:
         return
     # Disable logs going to Sentry. Its slow
-    sentry_sdk.init(integrations=[LoggingIntegration(level=None, event_level=None)])
+    sentry_sdk.init(integrations=[LoggingIntegration(level=None, event_level=None)], before_send=before_send)
 
 
 def raise_exception_with_sentry_if_available(
-    err: Exception, fingerprint: typing.Any
+    err: Exception
 ) -> typing.NoReturn:
     init_sentry()
```

Comment:
a word of caution about dealing with sentry sdk’s high-level apis (sentry_sdk.init in this case): if you use them as is, you might interfere with 3rd-party usages of sentry (they init it first, you reinit it, and start consuming their events). consider porting the relevant code in the sdk that uses lower-level api’s and also handles a few gotchas regarding e.g. sentry sessions, its async ops etc.
https://github.com/wandb/wandb/blob/main/wandb/analytics/sentry.py

Reply:
Wow - this is a super great call! Luckily, this code path is only in the weave service right now, where we control the packages, but even so, we should totally use your wrapper.